src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -403,11 +403,18 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
                         continue;
                     }
 
+                    if (functions.Count == ScriptConfig.MaximumFunctionCount)
+                    {
+                        AddFunctionError(functionName, string.Format(CultureInfo.InvariantCulture,
+                            "Maximun number of functions ({0}) exceeded. Ignoring function.", ScriptConfig.MaximumFunctionCount));
+                        continue;
```

Comment:
Don't we just want to `break` here? Emit a single log entry "The maximum number of functions (N) has been exceeded. Skipping the remaining (M) functions."